### PR TITLE
video response works, needs refactor

### PR DESCRIPTION
**Pull request by TohidHeshmati**
I know it is pretty ugly code. I did just monkey coded what ever you did for photo. it is ok to not accept this PR ;)
I followed the logic of old saying: "if it's stupid but it works, it is not stupid"

### CODE REVIEW

File: core/src/main/java/ninja/soroosh/chatopia/core/connectors/telegram/ResponseHandler.java [hunk 1]
```diff
@@ -40,59 +48,91 @@ public void init() {
...
-                }
-
-                MultiValueMap<String, Object> form = new LinkedMultiValueMap<>();
-                form.add("photo", fileEntity);
-                form.add("chat_id", chatId);
-                form.add("caption", photoResponse.getCaption());
-                HttpEntity<MultiValueMap<String, Object>> requestEntity
-                        = new HttpEntity<>(form, headers);
-
+                result = formGenerator(response, TYPE_PHOTO);
```
Comment (s-soroosh) [inline core/src/main/java/ninja/soroosh/chatopia/core/connectors/telegram/ResponseHandler.java]: extract the API call out from the formGenerator. 
let it to just prepare the request.

Comment (s-soroosh) [inline core/src/main/java/ninja/soroosh/chatopia/core/connectors/telegram/ResponseHandler.java]: genenerateForm  maybe?

File: core/src/main/java/ninja/soroosh/chatopia/core/connectors/telegram/ResponseHandler.java [hunk 2]
```diff
@@ -40,59 +48,91 @@ public void init() {
     //    TODO: model a general handling return type
     public Object handle(TelegramMessage message, Response response) {
         Object result = null;
-        final long chatId = message.getChat().getId();
+        chatId = message.getChat().getId();
```
Comment (s-soroosh) [inline core/src/main/java/ninja/soroosh/chatopia/core/connectors/telegram/ResponseHandler.java]: Don't make it class field. it's different from a message to another one

File: examples/src/main/java/ninja/soroosh/chatopia/core/example/MessageEchoExample.java [hunk 3]
```diff
@@ -62,6 +62,11 @@ public Response onPhCommand(String message, Context context) throws MalformedURL
                 .withCaption("chetoram?");
     }
 
+    @OnCommand(value = "video", help = "A showcase to send a video")
+    public Response onVideoCommand(String message, Context context) {
+        return Response.asVideo(this.getClass().getClassLoader().getResourceAsStream("thankyounote.mp4"));
```
Comment (s-soroosh) [inline examples/src/main/java/ninja/soroosh/chatopia/core/example/MessageEchoExample.java]: maybe a smaller mp4 file? 
This one is ~9.5MB, a bit too big to clone with low speed internet.

File: core/src/main/java/ninja/soroosh/chatopia/core/runner/responses/Response.java [hunk 4]
```diff
@@ -19,4 +19,20 @@ static PhotoResponse asPhoto(URL url) {
...
+
+    static VideoResponse asVideo(URL url) {
+        return new VideoResponse(url);
+    }
+
+    static VideoResponse asVideo(String id) {
+        return new VideoResponse(id);
+    }
+
+    InputStream getStream();
```
Comment (s-soroosh) [inline core/src/main/java/ninja/soroosh/chatopia/core/runner/responses/Response.java:35]: Why is this needed at the interface level?

Comment (TohidHeshmati) [reply core/src/main/java/ninja/soroosh/chatopia/core/runner/responses/Response.java:35]: instead of using `getVideoStream()` I wanted to use something more abstract so in future it would support other messages as file.

Comment (s-soroosh) [reply core/src/main/java/ninja/soroosh/chatopia/core/runner/responses/Response.java:35]: Right, but no every Response has stream, e.g. TextResponse
